src/VecSim/vec_sim.cpp

```diff
@@ -8,6 +8,63 @@
 #include <cassert>
 #include "memory.h"
 
+static VecSimResolveCode _ResolveParams_EFRuntime(VecSimAlgo index_type, VecSimRawParam rparam,
+                                                  VecSimQueryParams *qparams, bool hybrid,
+                                                  size_t k) {
+    long long num_val;
+    // EF_RUNTIME is a valid parameter only in HNSW algorithm.
+    if (index_type != VecSimAlgo_HNSWLIB) {
+        return VecSimParamResolverErr_UnknownParam;
+    }
+    if (qparams->hnswRuntimeParams.efRuntime != 0) {
+        return VecSimParamResolverErr_AlreadySet;
+    }
+    if (validate_positive_integer_param(rparam, &num_val) != VecSimParamResolver_OK) {
+        return VecSimParamResolverErr_BadValue;
+    }
+    // Asking for EF_RUNTIME lower than k is useless for standard KNN, as HNSW sets EF_RUNTIME to be
+    // >= k
+    if (k > (size_t)num_val && !hybrid) {
+        return VecSimParamResolverErr_k_GT_EfRuntime;
+    }
```

Comment:
breaking change?

Reply:
Yes... should avoid? If so, we'll loose the consistency from hybrid query (where we don't allow `ef_runtime < bacth_size`)